xenstoreclient/xenstore.go

```diff
@@ -529,7 +536,23 @@ func (xs *CachedXenStore) StopWatch() error {
 }
 
 func (xs *CachedXenStore) Clear() {
-	xs.writeCache = make(map[string]string, 0)
+	xs.writeCache = make(map[string]ValueUpdate, 0)
+}
+
+func (xs *CachedXenStore) UpdateCache() error {
+	for key, value := range xs.writeCache {
+		if value.update {
+			value.update = false
+			xs.writeCache[key] = value
+		} else {
+			delete(xs.writeCache, key)
+			err := xs.Rm(key)
+			if err != nil {
```

Comment:
`xenstore-rm` source code:
```
static int
do_rm(char *path, struct xs_handle *xsh, xs_transaction_t xth)
{
    if (xs_rm(xsh, xth, path)) {
        return 0;
    }
    else {
        warnx("could not remove path %s", path);
        return 1;
    }
}
```
The Error return code not detailed.

So, change like this:
```
		else {
			err := xs.Rm(key)
			if err != nil {
				return err
			} else {
				delete(xs.writeCache, key)
			}
		}
```